### ISSUE TITLE
Unexpected failure on emulated pairing check

### ISSUE THREAD

**Issue by hussein-aitlahcen** [Consensys/gnark#807]
Hi, 

I quickly hit an issue when trying to do many different pairing checks and can reproduce an issue where the `addStep` called in the miller loop is failing with something that look like a zero point addition:

```
                       fields_bn254.Ext2.AssertIsEqual
                                        e2.go:292
                                fields_bn254.Ext2.DivUnchecked
                                        e2.go:338
                                sw_bn254.Pairing.addStep
                                        pairing.go:583
                                sw_bn254.Pairing.MillerLoop
                                        pairing.go:353
                                sw_bn254.Pairing.Pair
                                        pairing.go:207
```

If I patch [addStep](https://github.com/Consensys/gnark/blob/bd4a39719a964f0305ee9ec36b6226e4c266584c/std/algebra/emulated/sw_bn254/pairing.go#L652) with a variant similar to [AddUnified](https://github.com/Consensys/gnark/blob/bd4a39719a964f0305ee9ec36b6226e4c266584c/std/algebra/emulated/sw_emulated/point.go#L158) the error disappear and the pairing check is passing.
Is this a corner case we don't handle in gnark yet, i.e. can the miller loop hit zero point when doing the addition?

**Comment by gbotrel** [comment]
cc @yelhousni

**Comment by yelhousni** [comment]
What test vector did you use to hit this issue? in the error below `addStep` is used in the second iteration of the Miller loop to replace `doubleAndAddStep(Qacc,-Q)` (`2Qacc-Q`) by `addStep(Qacc,Q)` (`Qacc+Q`). At this iteration, `Qacc=2Q` so edge cases should not happen in the addition (no need to use `addUnifedStep`). Maybe you provided a point `Q=0`?

**Comment by hussein-aitlahcen** [comment]
I no longer have the regression test (my bad really...). But I know we were on a pretty outdated develop commit prior to https://github.com/Consensys/gnark/pull/603. Maybe #603 solved it (I will see if I hit the issue again without the addStepUnified). Thanks for investigating :). I don't think we were having a zero point in `Q` as the pairing check would have failed in the end? Also I noticed the gnark-crypto pairing algorithm do use an addition that handle the zero case, even in the miller loop, thats why I went in this direction. I'll let you know anyway, thanks for the help!
